src/Compilers/CSharp/Test/Syntax/Parsing/ParserErrorMessageTests.cs

```diff
@@ -3112,53 +3112,47 @@ public static class GenExtensions<X> where X : struct
-                // (6,34): error CS8339:  The parameter modifier 'ref' cannot be used after the modifier 'this' 
-                //     public static void Goo1(this ref int i) {}
-                Diagnostic(ErrorCode.ERR_BadParameterModifiersOrder, "ref").WithArguments("ref", "this").WithLocation(6, 34),
-                // (18,37): error CS8339:  The parameter modifier 'ref' cannot be used after the modifier 'this' 
-                //     public static void Goo2<T>(this ref T t) {}
-                Diagnostic(ErrorCode.ERR_BadParameterModifiersOrder, "ref").WithArguments("ref", "this").WithLocation(18, 37),
                 // (12,21): error CS1106: Extension method must be defined in a non-generic static class
-                // public static class GenExtensions<X>
+                // public static class GenExtensions<X> where X : struct
                 Diagnostic(ErrorCode.ERR_BadExtensionAgg, "GenExtensions").WithLocation(12, 21),
-                // (19,37): error CS8339:  The parameter modifier 'ref' cannot be used after the modifier 'this' 
-                //     public static void Goo2<T>(this ref X x) {}
-                Diagnostic(ErrorCode.ERR_BadParameterModifiersOrder, "ref").WithArguments("ref", "this").WithLocation(19, 37),
                 // (12,21): error CS1106: Extension method must be defined in a non-generic static class
-                // public static class GenExtensions<X>
+                // public static class GenExtensions<X> where X : struct
                 Diagnostic(ErrorCode.ERR_BadExtensionAgg, "GenExtensions").WithLocation(12, 21),
-                // (21,41): error CS8339:  The parameter modifier 'ref' cannot be used after the modifier 'this' 
-                //     public static void Goo2<T,U,V>(this ref U u) {}
-                Diagnostic(ErrorCode.ERR_BadParameterModifiersOrder, "ref").WithArguments("ref", "this").WithLocation(21, 41),
                 // (12,21): error CS1106: Extension method must be defined in a non-generic static class
-                // public static class GenExtensions<X>
+                // public static class GenExtensions<X> where X : struct
                 Diagnostic(ErrorCode.ERR_BadExtensionAgg, "GenExtensions").WithLocation(12, 21),
-                // (15,34): error CS8339:  The parameter modifier 'ref' cannot be used after the modifier 'this' 
-                //     public static void Goo2(this ref int i) {}
-                Diagnostic(ErrorCode.ERR_BadParameterModifiersOrder, "ref").WithArguments("ref", "this").WithLocation(15, 34),
                 // (12,21): error CS1106: Extension method must be defined in a non-generic static class
-                // public static class GenExtensions<X>
+                // public static class GenExtensions<X> where X : struct
                 Diagnostic(ErrorCode.ERR_BadExtensionAgg, "GenExtensions").WithLocation(12, 21));
         }
 
+        [Fact]
+        public void BadParameterModifiers()
```

Comment:
>BadParameterModifiers [](start = 20, length = 21)

Can we change the test name to describe which modifiers are bad?

Reply:
I suggest adding the opposite test scenarios as well:

* Testing we produce an error on `in this ref int i`
* Testing we produce an error on `in this in int i`

---
In reply to: [157594285](https://github.com/dotnet/roslyn/pull/23643#discussion_r157594285) [](ancestors = 157594285)